api-operator/pkg/apim/deleteAPI.go

```diff
@@ -0,0 +1,135 @@
+			logDelete.Info("APIM config is not found. Continue with default configs")
+			return errInput
+		} else {
+			logDelete.Error(errInput, "Error retrieving APIM configs")
+			return errInput
+		}
+	}
+
+	kmEndpoint := apimConfig.KeyManagerEndpoint
+	publisherEndpoint := apimConfig.PublisherEndpoint
+	tokenEndpoint := apimConfig.TokenEndpoint
+	credSecret := apimConfig.CredentialsSecretName
+
+	insecure = apimConfig.SkipVerification
+
+	if strings.EqualFold(tokenEndpoint, "") {
+		tokenEndpoint = kmEndpoint + "/" + defaultTokenEndpoint
+		logDelete.Info("Token endpoint not defined. Using keymanager endpoint.", "tokenEndpoint", tokenEndpoint)
+	}
+
+	accessToken, errToken := getAccessToken(client, tokenEndpoint, kmEndpoint, credSecret)
+	if errToken != nil {
+		return errToken
+	}
+
+	if inputConf.BinaryData != nil {
+		deleteErr := deleteAPIFromProject(inputConf, accessToken, publisherEndpoint)
+		if deleteErr != nil {
+			logDelete.Error(deleteErr, "Error when deleting the API using zip")
+			return deleteErr
+		}
+	} else {
+		deleteErr := deleteAPIFromSwagger(inputConf, accessToken, publisherEndpoint)
+		if deleteErr != nil {
+			logDelete.Error(deleteErr, "Error when deleting the API using swagger")
+			return deleteErr
+		}
+	}
+
+	return nil
+}
+
+func deleteAPIFromProject(config *corev1.ConfigMap, token string, endpoint string) error {
+	zipFileName, errZip := maps.OneKey(config.BinaryData)
+	if errZip != nil {
+		return errZip
+	}
+	zippedData := config.BinaryData[zipFileName]
+
+	tmpPath, err := getTempPathOfExtractedArchive(zippedData)
+	if err != nil {
+		logDelete.Error(err, "Error while getting extracted temporary directory")
+		return err
+	}
+
+	// Get API info
+	apiInfo, err := getAPIDefinition(tmpPath)
+	if err != nil {
+		logDelete.Error(err, "Error while getting API definition")
+		return err
+	}
+
+	// checks whether the API exists in APIM
+	apiId, err := getAPIId(token, endpoint+"/"+defaultApiListEndpointSuffix, apiInfo.ID.APIName, apiInfo.ID.Version)
+	if err != nil {
+		return err
+	}
+
+	deleteErr := deleteAPIById(endpoint, apiId, token)
+	if deleteErr != nil {
+		logDelete.Error(deleteErr, "Error when deleting the API from APIM")
+	}
+
+	return nil
+}
+
+func deleteAPIFromSwagger(config *corev1.ConfigMap, token string, endpoint string) error {
+	swaggerFileName, errSwagger := maps.OneKey(config.Data)
+	if errSwagger != nil {
+		logImport.Error(errSwagger, "Error in the swagger configmap data", "data", config.Data)
+		return errSwagger
+	}
+	swaggerData := config.Data[swaggerFileName]
+
+	_, name, version, err := getAdditionalProperties(swaggerData)
+	if err != nil {
+		logImport.Error(err, "Error getting additional data")
+		return err
+	}
+
+	apiId, err := getAPIId(token, endpoint+"/"+defaultApiListEndpointSuffix, name, version)
+	if err != nil {
+		return err
+	}
+
+	deleteErr := deleteAPIById(endpoint, apiId, token)
+	if deleteErr != nil {
+		logDelete.Error(deleteErr, "Error when deleting the API from APIM")
```

Comment:
Shouldn't this error be passed to the caller

Reply:
Fixed in e0400b3